### PR TITLE
feat: add aria live to language mode button

### PR DESCRIPTION
**Pull request by henriettemoe**

### CODE REVIEW

File: language/nn.json [hunk 1]
```diff
@@ -90,6 +90,38 @@
...
           "label": "Endret svarmodus aria",
           "default": "Endret svarmodus til @option"
+        },
+        {
+          "label": "Aria for språkmodus",
+          "default": "Byt språk frå @sourceLanguage til @targetLanguage"
+        },
+        {
+          "label": "Endret språkmodus aria",
+          "default": "Bytet språk frå @sourceLanguage til @targetLanguage"
```
Comment (boyum) [inline language/nn.json:100]:
```suggestion
          "default": "Bytte språk frå @sourceLanguage til @targetLanguage"
```

File: src/types/types.ts [hunk 2]
```diff
@@ -16,6 +17,8 @@ export enum LanguageModeType {
   Target = 'target',
 }
 
+export type LanguageCode = (typeof languages)[number];
```
Comment (boyum) [inline src/types/types.ts:20]: Unless we're actually using `const languages`, we could do this to infer LanguageCode from semantics: `export type LanguageCode = Params['sourceLanguage'];`

File: src/utils/language.utils.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+export const getLanguageModeAria = (
+  languageMode: LanguageModeType,
+  ariaText: string,
+  sourceLanguageCode: string,
+  targetLanguageCode: string
+): string => {
+  const { t } = useTranslation();
+  const sourceLanguage = t(`lang_${sourceLanguageCode as LanguageCode}`);
+  const targetLanguage = t(`lang_${targetLanguageCode as LanguageCode}`);
```
Comment (boyum) [inline src/utils/language.utils.ts:12]:
```suggestion
  sourceLanguageCode: LanguageCode,
  targetLanguageCode: LanguageCode,
): string => {
  const { t } = useTranslation();
  const sourceLanguage = t(`lang_${sourceLanguageCode}`);
  const targetLanguage = t(`lang_${targetLanguageCode}`);
```

File: src/types/types.ts (hunk 2 above)
Comment (henriettemoe) [reply src/types/types.ts:20]: Nice!

Comment (henriettemoe) [reply src/types/types.ts]: Done ✔️

**Comment by henriettemoe** [comment]
The aria currently reads that the user can "swap from languageX to languageY", where the language is the target language that the user must fill or drag in place. Could perhaps have a different text that explains that this is in fact the target language.

**Review by boyum — APPROVED** [review]

**Pull request merged**